netlify/functions/manage-findings.ts

```diff
@@ -0,0 +1,428 @@
+    edited = true;
+  }
+
+  if (!edited) {
+    // throw?
+    // {
+    //   status: 204,
+    //   body: "Nothing changed",
+    // };
+    return;
+  }
+
+  await client.request(
+    "POST /repos/{owner}/{repo}/issues/{issue_number}/labels",
+    {
+      owner: process.env.GITHUB_REPO_OWNER!,
+      repo: repoName,
+      issue_number: issueNumber,
+      labels: ["edited-by-warden"],
+    }
+  );
+
+  if (apiKey && domain && process.env.EMAIL_SENDER) {
+    const subject = `C4 ${contest.sponsor} finding updated`;
+
+    await sendConfirmationEmail(data.emailAddresses, subject, emailBody);
+  }
+}
+
+const handler: Handler = async (event: Event): Promise<Response> => {
+  // todo: error handling..
+
+  if (!(await checkAuth(event))) {
+    return {
+      statusCode: 401,
+      body: JSON.stringify({
+        error: "Unauthorized",
+      }),
+    };
+  }
+
+  // should contestId always be included in query params for consistency?
+  let contestId;
+  if (event.queryStringParameters?.contest) {
+    contestId = parseInt(event.queryStringParameters?.contest);
+  } else if (event.body) {
+    const data = JSON.parse(event.body);
+    contestId = data.contest;
+  }
+
+  // make sure the given contest is still active
+  const contest = await getContest(contestId);
+  if (!contest) {
+    return {
+      statusCode: 422,
+      body: JSON.stringify({
+        error: "Contest not found",
+      }),
+    };
+  }
+  if (!isContestActive(contest)) {
+    return {
+      statusCode: 400,
+      body: JSON.stringify({
+        error: "This contest has ended.",
+      }),
+    };
+  }
+
+  // simple param filling
+  let username;
+  if (event.headers["c4-user"]) {
+    username = event.headers["c4-user"];
+  }
+
+  switch (event.httpMethod) {
+    case "GET":
+      let issueNumber;
+      if (event.queryStringParameters?.issue) {
+        issueNumber = parseInt(event.queryStringParameters?.issue);
+      }
+
+      let includeTeams = true;
+      // @todo: use flag to include teams in query params
+      // if (req.queryStringParameters?.includeTeams) {
+      // includeTeams = req.queryStringParameters?.includeTeams)
+      // }
+
+      if (issueNumber !== undefined) {
+        return await getFinding(username, contest, issueNumber);
+      } else {
+        return await getFindings(username, contest, includeTeams);
+      }
+    case "POST":
+      const data: FindingEditRequest = JSON.parse(event.body!);
+      try {
+        await editFinding(username, contest, data.issue, data);
+        return {
+          statusCode: 200,
+          body: "SUCCESS!",
+        };
+      } catch (error) {
+        return {
+          statusCode: error.status || 500,
+          body: JSON.stringify({
+            error: error.message || "something went wrong editing submission",
+          }),
+        };
+      }
+  }
+
+  return {
+    statusCode: 418,
+    body: JSON.stringify({
+      error: "nuh-uh",
```

Comment:
Lol, should we change the error message? Or will changing the status code be clear enough?

Reply:
🤷🫖

We'll refine states as we go 🙂